python/veles/schema/fields.py

```diff
@@ -191,6 +191,17 @@ def cpp_type(self):
 class String(Field):
     value_type = six.text_type
 
+    def __set__(self, instance, value):
+        if isinstance(value, str):
+            value = six.text_type(value)
+        super(String, self).__set__(instance, value)
+
+    def _dump(self, value):
+        return six.text_type(value)
+
+    def _validate(self, value):
```

Comment:
No.

The main problem with Python 2 is that unicode and str sort of morph into each other and you can never quite tell if you're dealing with one or the other.  This class is specifically designed to reject str on Python 2, to make sure we're always dealing with unicode.

Your changes make it possible to create a model instance with a String field set to either str or unicode (by unpacking it from msgpack), thrashing the type guarantees that are the whole point of the Model system.  This is caught by the unittests, which are (predictably) broken.